core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/PythonProxyClient.scala

```diff
@@ -158,12 +156,18 @@ class PythonProxyClient(portNumberPromise: Promise[Int], val actorId: ActorVirtu
   }
 
   override def close(): Unit = {
-
     val action: Action = new Action("shutdown")
-    flightClient.doAction(action) // do not expect reply
-
-    flightClient.close()
-
+    try {
+      flightClient.doAction(action) // do not expect reply
+
+      flightClient.close()
+    } catch {
+      case _: NullPointerException =>
+        running = false
+        logger.warn(
+          s"Failed to close since the flight client is null"
+        )
```

Comment:
change the log message? when flightClient is null I think it means it has no flightCliented created, thus no need to close it.